app/webpacker/components/Persons/Person/index.jsx

```diff
@@ -0,0 +1,177 @@
+import PersonalRecords from './PersonalRecords';
+import I18n from '../../../lib/i18n';
+import RegionalRecords from './RegionalRecords';
+import RegionalChampionshipPodiums from './RegionalChampionshipPodiums';
+import CompetitionsMap from './CompetitionsMap';
+import Results from './Results';
+import CountStats from './CountStats';
+
+function TabSection({
+  results,
+  records,
+  competitions,
+  championshipPodiums,
+  pbMarkers,
+  highlight,
+}) {
+  const panes = useMemo(() => {
+    const p = [{
+      menuItem: I18n.t('persons.show.results'),
+      tabSlug: 'results-by-event',
+      render: () => (
+        <Tab.Pane>
+          <Results
+            results={results}
+            pbMarkers={pbMarkers}
+            highlightPosition={highlight}
+            competitions={competitions}
+          />
+        </Tab.Pane>
+      ),
+    }];
+    if (records.total > 0) {
+      p.push({
+        menuItem: I18n.t('persons.show.records'),
+        tabSlug: 'record',
+        render: () => (
+          <Tab.Pane>
+            <RegionalRecords results={results} competitions={competitions} />
+          </Tab.Pane>
+        ),
+      });
+    }
+
+    const anyPodiums = Object
+      .values(championshipPodiums)
+      .some((podiums) => podiums.length > 0);
+
+    if (anyPodiums) {
+      p.push({
+        menuItem: I18n.t('persons.show.championship_podiums'),
+        tabSlug: 'championship-podiums',
+        render: () => (
+          <Tab.Pane>
+            <RegionalChampionshipPodiums
+              results={results}
+              championshipPodiums={championshipPodiums}
+              competitions={competitions}
+            />
+          </Tab.Pane>
+        ),
+      });
+    }
+
+    const competitionValues = Object.values(competitions);
+
+    p.push({
+      menuItem: I18n.t('persons.show.competitions_map'),
+      tabSlug: 'map',
+      render: () => (
+        <Tab.Pane>
+          <CompetitionsMap competitions={competitionValues} />
+        </Tab.Pane>
+      ),
+    });
+    return p;
+  }, [records, championshipPodiums, competitions, results, pbMarkers, highlight]);
+
+  const tabSlug = new URL(document.location.toString()).searchParams.get('tab');
+  const activeIndex = tabSlug ? panes.findIndex((p) => p.tabSlug === tabSlug) : 0;
+
+  return (
+    <Tab
+      renderActiveOnly
+      defaultActiveIndex={activeIndex}
+      panes={panes}
+      menu={{ fluid: true, widths: panes.length }}
+      onTabChange={(a, b) => {
+        const newSlug = panes[b.activeIndex].tabSlug;
+        const url = new URL(window.location.href);
+        url.searchParams.set('tab', newSlug);
+        window.history.pushState({}, '', url);
+      }}
+    />
+  );
+}
+
+export default function Person({
+  person,
+  previousPersons,
+  results,
+  averageRanks,
+  singleRanks,
+  competitions,
+  medals,
+  records,
+  pbMarkers,
+  championshipPodiums,
+  canEditUser,
+}) {
+  const [highlight, setHighlight] = useState(-1);
```

Comment:
Not sure what this is just from looking at this file, is there a more descriptive name?